src/coreclr/jit/gentree.h

```diff
@@ -1761,6 +1761,9 @@ struct GenTree
     inline bool IsVectorAllBitsSet() const;
     inline bool IsVectorBroadcast(var_types simdBaseType) const;
     inline bool IsMaskAllBitsSet() const;
+    inline bool IsMaskAllBitsZero() const;
+    inline bool IsAllTrue() const;
+    inline bool IsAllFalse() const;
```

Comment:
Are these needed? They should be the same as `AllBitsSet` and `Zero`, respectively?

Reply:
`IsAllTrue()` includes `IsVectorAllBitsSet`, while `IsAllFalse()` also includes `IsVectorZero()`.

Reply:
Yes, but we never have a case that can have both. For any given node we expect either a mask or a vector and have a precise API to call.

Reply:
Alright, so i was testing scenario, where we can have one of these:

```
Sve.ConditionalSelect(Vector<Double>.Zero, Sve.Abs(left), Sve.Abs(right));
Sve.ConditionalSelect(Sve.CreateFalseMaskDouble(), Sve.Abs(left), Sve.Abs(right));
```

Reply:
May be the first one should get convert to Mask by `CvtVectorToMask(Zero)`. Need to check what node we end up with that.

Reply:
> May be the first one should get convert to Mask by CvtVectorToMask(Zero)

Right. If the underlying `NI_Sve_ConditionalSelect` node is expecting `TYP_MASK, TYP_SIMD, TYP_SIMD` then op1 should always be `TYP_MASK` and any `TYP_SIMD` node in this position would be illegal, the IR should've inserted a `CvtVectorToMask` node in this position to wrap such a SIMD parameter

The same goes for op2/op3, it would be illegal for them to be `TYP_MASK` and any IR should've inserted a `CvtMaskToVector` node to wrap such a MASK parameter